gtsfm/utils/homography_decomposition.py

```diff
@@ -0,0 +1,384 @@
+"""
+Decompose an homography matrix into the possible rotations, translations,
+and plane normal vectors.
+
+Based off of homography decomposition implementation in COLMAP:
+    https://github.com/colmap/colmap/blob/dev/src/base/homography_matrix.cc
+    https://github.com/colmap/colmap/blob/dev/src/base/homography_matrix.h
+See how it is used in COLMAP:
+https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L198
+
+COLMAP and OpenCV's implementationsare based off of:
+Ezio Malis, Manuel Vargas, and others. Deeper understanding of the homography decomposition for vision-based control. 2007.
+https://hal.inria.fr/inria-00174036/PDF/RR-6303.pdf
+
+OpenCV does not support the case of intrinsics from two separate cameras, however.
+
+Authors: John Lambert (Python), from original C++
+"""
+
+from typing import List, Tuple
+
+import numpy as np
+from gtsam import Rot3, Unit3
+
+"""
+PoseFromHomographyMatrix(
+        H, camera1.CalibrationMatrix(), camera2.CalibrationMatrix(),
+        inlier_points1_normalized, inlier_points2_normalized, &R, &tvec, &n,
+        &points3D);
+
+  if (points3D.empty()) {
+    tri_angle = 0;
+  } else {
+    tri_angle = Median(CalculateTriangulationAngles(
+        Eigen::Vector3d::Zero(), -R.transpose() * tvec, points3D));
+  }
+
+  if (config == PLANAR_OR_PANORAMIC) {
+    if (tvec.norm() == 0) {
+      config = PANORAMIC;
+      tri_angle = 0;
+    } else {
+      config = PLANAR;
+    }
+"""
+
+
+def pose_from_homography_matrix(
+    H: np.ndarray,
+    K1: np.ndarray,
+    K2: np.ndarray,
+    points1: np.ndarray,
+    points2: np.ndarray,
```

Comment:
why not use keypoints? It maintains consistency.

Reply:
I guess I was thinking this was a bit unusual, since we usually use Keypoints for the whole set of detected features in an image, but these are just the subset of correspondences.